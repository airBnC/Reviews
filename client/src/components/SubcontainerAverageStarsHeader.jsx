import React from 'react';
import PropTypes from 'prop-types';

import TotalReviews from './TotalReviews';
import SearchReviews from './SearchReviews';

import { Wrapper } from './styles/SubcontainerAverageStarsHeader.styles';

const SubcontainerAverageStarsHeader = props => (
  <Wrapper>
    <TotalReviews
      overallAvg={props.overallAvg}
      numReviews={props.numReviews}
    />
    <SearchReviews />
  </Wrapper>
);

SubcontainerAverageStarsHeader.propTypes = {
  overallAvg: PropTypes.number,
  numReviews: PropTypes.number,
};

SubcontainerAverageStarsHeader.defaultProps = {
  overallAvg: 1,
  numReviews: 1,
};

export default SubcontainerAverageStarsHeader;
