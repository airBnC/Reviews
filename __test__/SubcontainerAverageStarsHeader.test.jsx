import React from 'react';
import { configure, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import SubcontainerAverageStarsHeader from '../client/src/components/SubcontainerAverageStarsHeader';

configure({ adapter: new Adapter() });

const dummyProps = {
  overallAvg: 3.4,
  numReviews: 18,
  currentSearch: 'pineapples',
};

describe('SubcontainerAverageStarsHeader', () => {
  it('should render correctly', () => {
    const output = shallow(<SubcontainerAverageStarsHeader
      overallAvg={dummyProps.overallAvg}
      numReviews={dummyProps.numReviews}
      currentSearch={dummyProps.currentSearch}
    />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
