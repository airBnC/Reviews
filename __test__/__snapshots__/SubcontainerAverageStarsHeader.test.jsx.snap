// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubcontainerAverageStarsHeader should render correctly 1`] = `
<styled.div>
  <TotalReviews
    numReviews={18}
    overallAvg={3.4}
  />
  <SearchReviews
    currentSearch="pineapples"
    handleSearchChange={[Function]}
    handleSearchSubmit={[Function]}
  />
</styled.div>
`;
